e2e_playwright/app_hotkeys_test.py

```diff
@@ -41,11 +43,11 @@ def test_does_not_clear_cache_dialog_when_c_is_pressed_inside_text_input(app: Pa
 
 def test_reruns_when_r_is_pressed(app: Page):
     app.keyboard.type("r")
-    expect(app.get_by_test_id("stStatusWidget")).to_be_visible()
+    expect_script_state(app, "running")
```

Comment:
has this ever caused flakiness? If the rerun happens fast enough, the status will never be observed, right?
I am wondering whether we should use the same `MutationObserver` logic that we added in https://github.com/streamlit/streamlit/pull/9126 exactly for this reason.

Reply:
Nope, not that I know of. But we could potentially migrate to a similar behavior with `MutationObserver` if we make the underlying logic a bit more generic.